File: src/algorithms/can-be-increasing.js
/**
 * Given a 0-indexed integer array nums, returns true if it can be made
 * strictly increasing after removing exactly one element, or false otherwise.
 * If the array is already strictly increasing, returns true.
 *
 * The array nums is strictly increasing if nums[i - 1] < nums[i] for each
 * index (1 <= i < nums.length).
 *
 * @format
 * @param {number[]} nums - An array of numbers
 * @returns {boolean} - Returns true if the array is strictly increasing after
 *   removing one or fewer elements
 */

function canBeIncreasing(nums) {
  let count = 0;
  for (let i = 1; i < nums.length && count < 2; i++) {
    if (nums[i - 1] >= nums[i]) {
      count++;
      if (i > 1 && nums[i - 2] >= nums[i]) {
        nums[i] = nums[i - 1];
      }
    }
  }
  return count < 2;
}

File: src/algorithms/can-be-increasing.py
def canBeIncreasing(nums):
    """Given a 0-indexed integer array nums, return true if it can be made 
    strictly increasing after removing exactly one element, or false otherwise. 
    If the array is already strictly increasing, return true. The array nums is 
    strictly increasing if nums[i - 1] < nums[i] for each index (1 <= i < nums.
    length).

    Args:
        nums (list[int]): A list of positive integers.

    Returns:
        Boolean: Returns True if array can be made strictly increasing.
    """
    count = 0
    for i in range(1, len(nums)):
        if nums[i - 1] >= nums[i]:
            count += 1
            if i > 1 and nums[i - 2] >= nums[i]:
                nums[i] = nums[i - 1]
        if count >= 2:
            break
    return count < 2
